build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets

```diff
@@ -0,0 +1,40 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <Target Name="GenerateNuGetPackagesArchive"
+            DependsOnTargets="SetupNuGetPackagesArchiveInputsOutputs"
+            Inputs="$(Stage2Directory)/sdk/$(SdkVersion)/dotnet.dll"
+            Outputs="$(FinalArchive)">
+      <ItemGroup>
+        <FilesToClean Include="$(NuGetPackagesArchiveProject)/**/*" />
+        <FilesToClean Include="$(NuGetPackagesArchiveFolder)/**/*" />
+      </ItemGroup>
+
+      <Delete Files="@(FilesToClean)" />
+      <RemoveDir Directories="$(NuGetPackagesArchiveProject);$(NuGetPackagesArchiveFolder)" />
+      <MakeDir Directories="$(NuGetPackagesArchiveProject);$(NuGetPackagesArchiveFolder)"/>
+
+      <Exec Command="$(DotnetStage2) new" WorkingDirectory="$(NuGetPackagesArchiveProject)" />
+      <Exec Command="$(DotnetStage2) restore --packages $(NuGetPackagesArchiveFolder)"
+          WorkingDirectory="$(NuGetPackagesArchiveProject)" />
+
+      <Delete Files="$(IntermediateArchive);$(IntermediateArchive).zip" />
+
+      <Message Text="Publishing Archiver" />
+      <Exec Command="$(DotnetStage2) publish --output $(ToolsOutputDirectory) --configuration $(Configuration)" 
+          WorkingDirectory="$(RepoRoot)/tools/Archiver" />
+
+      <Exec Command="$(ArchiverExe) -a $(IntermediateArchive) $(NuGetPackagesArchiveFolder)" />
+
+      <Copy SourceFiles="$(IntermediateArchive)" DestinationFiles="$(FinalArchive)" />
```

Comment:
Why do we have an "IntermediateArchive"?  Why not just tell the ArchiverExe to put it in the FinalArchive destination in the first place?